src/theory/model_manager.h

```diff
@@ -0,0 +1,114 @@
+/*********************                                                        */
+/*! \file model_manager.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Abstract management of models for TheoryEngine.
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef CVC4__THEORY__MODEL_MANAGER__H
+#define CVC4__THEORY__MODEL_MANAGER__H
+
+#include <map>
+#include <memory>
+
+#include "theory/theory_model.h"
+#include "theory/theory_model_builder.h"
+
+namespace CVC4 {
+
+class TheoryEngine;
+
+namespace theory {
+
+/**
+ * A base class for managing models. Its main feature is to implement a
+ * buildModel command, which can be specific e.g. to the kind of equality
+ * engine management mode we are using.
```

Comment:
I am not quite following that comment. `buildModel` is (currently?) not actually specific to the kind of equality engine management.

Reply:
buildModel depends on prepareModel, which is overriden by ModelManagerDistributed.  I'll clarify this comment.